internal/handlers/msg_fee_invoker.go

```diff
@@ -79,10 +79,12 @@ func (afd MsgFeeInvoker) Invoke(ctx sdk.Context, simulate bool) (sdk.Coins, sdk.
 
 		// If there's fees left to collect, or there were consumed fees, deduct/distribute them now.
 		if !unchargedFees.IsZero() || !consumedFees.IsZero() {
+			ctx = ctx.WithEventManager(sdk.NewEventManager())
```

Comment:
The new context should be put into a new variable since we only want the fresh event manager for this if block.

Reply:
Done.